### PR TITLE
switch oauthserver to external user client

### PR DESCRIPTION
**Pull request by deads2k**
This switches the vast majority of calls to external user calls. There's a bad unit test that needs to move, but that is involved.

I also found some packages that are not used inside of the oauthserver and used instead in the apiserver.

/assign enj
/assign mfojtik

### CODE REVIEW

File: pkg/apiserver/authentication/internaloauth/expirationvalidator_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package internaloauth
```
Comment (enj) [inline pkg/apiserver/authentication/internaloauth/expirationvalidator_test.go]: Can you do this as a move to preserve the history of this file.

File: pkg/apiserver/authentication/internaloauth/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package internaloauth
```
Comment (enj) [inline pkg/apiserver/authentication/internaloauth/interfaces.go]: Same.

File: pkg/apiserver/authentication/internaloauth/tokenauthenticator_test.go [hunk 3]
```diff
@@ -0,0 +1,383 @@
+package internaloauth
```
Comment (enj) [inline pkg/apiserver/authentication/internaloauth/tokenauthenticator_test.go]: Same.

File: pkg/apiserver/authentication/internaloauth/expirationvalidator_test.go (hunk 1 above)
Comment (deads2k) [reply pkg/apiserver/authentication/internaloauth/expirationvalidator_test.go]: > Can you do this as a move to preserve the history of this file.

No.  The previous one part of a giant test file.  There is no previous file to move.

File: pkg/apiserver/authentication/internaloauth/tokenauthenticator_test.go (hunk 3 above)
Comment (deads2k) [reply pkg/apiserver/authentication/internaloauth/tokenauthenticator_test.go]: > Same.

No.  The previous one part of a giant test file.  There is no previous file to move.

File: pkg/apiserver/authentication/internaloauth/interfaces.go (hunk 2 above)
Comment (deads2k) [reply pkg/apiserver/authentication/internaloauth/interfaces.go]: > Same.

No.  The previous was a set of combined interfaces and git chose the other half of the interfaces as the move.

**Comment by deads2k** [comment]
I can't both move and not move a single file.  Half of those files remain in their original positions.

Tagging based on irc conversation.

**Pull request merged**